File: java/code program/Showroom.java
import java.util.ArrayList;
import java.util.List;

public class Showroom {
    private String nama;
    private String lokasi;
    private List<Truck> truk;
    private List<Motorcycle> motor;
    private List<Mobil> mobil;

    public Showroom(String nama, String lokasi) {
        this.nama = nama;
        this.lokasi = lokasi;
        this.truk = new ArrayList<>();
        this.motor = new ArrayList<>();
        this.mobil = new ArrayList<>();
    }

    public String getNama() {
        return nama;
    }

    public void setNama(String nama) {
        this.nama = nama;
    }

    public String getLokasi() {
        return lokasi;
    }

    public void setLokasi(String lokasi) {
        this.lokasi = lokasi;
    }

    public void addTruck(Truck t) {
        truk.add(t);
    }

    public void addMotorcycle(Motorcycle m) {
        motor.add(m);
    }

    public void addCar(Mobil m) {
        mobil.add(m);
    }

    public void printMotorcycles() {
        System.out.println("Daftar Motor di Showroom " + nama + ":");
        if (motor.isEmpty()) {
            System.out.println("Tidak ada motor di showroom.");
            return;
        }
        for (int i = 0; i < motor.size(); i++) {
            Motorcycle m = motor.get(i);
            System.out.println("Motor " + (i + 1) + ":");
            System.out.println("Brand: " + m.getBrand());
            System.out.println("Model: " + m.getModel());
            System.out.println("Year: " + m.getYear());
            System.out.println("Price: " + m.getPrice());
            System.out.println("Type: " + m.getType());
            System.out.println("Has Windshield: " + (m.hasWindshield() ? "Yes" : "No"));
            System.out.println("--------------------");
        }
    }

    public void printTrucks() {
        System.out.println("Daftar Truk di Showroom " + nama + ":");
        if (truk.isEmpty()) {
            System.out.println("Tidak ada truk di showroom.");
            return;
        }
        for (int i = 0; i < truk.size(); i++) {
            Truck t = truk.get(i);
            System.out.println("Truk #" + (i + 1) + ":");
            System.out.println("Brand: " + t.getBrand());
            System.out.println("Model: " + t.getModel());
            System.out.println("Year: " + t.getYear());
            System.out.println("Price: " + t.getPrice());
            System.out.println("Kapasitas: " + t.getKapasitas() + " ton");
            System.out.println("Gerbong: " + (t.isGerbong() ? "Yes" : "No"));
            System.out.println("--------------------");
        }
    }

    public void printCars() {
        System.out.println("Daftar Mobil di Showroom " + nama + ":");
        if (mobil.isEmpty()) {
            System.out.println("Tidak ada mobil di showroom.");
            return;
        }
        for (int i = 0; i < mobil.size(); i++) {
            Mobil m = mobil.get(i);
            System.out.println("Mobil #" + (i + 1) + ":");
            System.out.println("Brand: " + m.getBrand());
            System.out.println("Model: " + m.getModel());
            System.out.println("Year: " + m.getYear());
            System.out.println("Price: " + m.getPrice());
            System.out.println("Doors: " + m.getDoors());
            System.out.println("Fuel Type: " + m.getFuelType());
            System.out.println("--------------------");
        }
    }

    public void printAllVehicles() {
        System.out.println("Showroom: " + nama + " | Lokasi: " + lokasi);
        System.out.println("=================================");
        printMotorcycles();
        System.out.println();
        printTrucks();
        System.out.println();
        printCars();
        System.out.println("=================================");
    }
}

File: cpp/code program/showroom.cpp
#pragma once
#include <bits/stdc++.h> // library standar dari C++
#include "kendaraan.cpp"
#include "motor.cpp"
#include "mobil.cpp"
#include "truck.cpp"

using namespace std;

class Showroom {
private:
    string nama, lokasi;
    vector<Truck> truk;
    vector<Motorcycle> motor;
    vector<Mobil> mobil;

public:
    // Constructor
    Showroom(string n, string l) {
        this->nama = n;
        this->lokasi = l;
    }

    // Getter
    string getNama() { return this->nama; }
    string getLokasi() { return this->lokasi; }
    vector<Truck> getTrucks() { return this->truk; }
    vector<Motorcycle> getMotorcycles() { return this->motor; }
    vector<Mobil> getCars() { return this->mobil; }

    // Setter
    void setNama(string n) { nama = n; }
    void setLokasi(string l) { lokasi = l; }

    // Menambahkan kendaraan ke showroom
    void addTruck(Truck t) { truk.push_back(t); }
    void addMotorcycle(Motorcycle m) { motor.push_back(m); }
    void addCar(Mobil m) { mobil.push_back(m); }

    // Cetak daftar motor
    void printMotorcycles() {
        cout << "Daftar Motor di Showroom " << nama << ":\n";
        if (motor.empty()) {
            cout << "Tidak ada motor di showroom.\n";
            return;
        }
        for (int i = 0; i < motor.size(); i++) {
            cout << "Motor " << i + 1 << ":\n";
            cout << "Brand : " << motor[i].getbrand() << endl;
            cout << "Model : " << motor[i].getmodel() << endl;
            cout << "Year : " << motor[i].getyear() << endl;
            cout << "Price : " << motor[i].getprice() << endl;
            cout << "Type : " << motor[i].getType() << endl;
            cout << "Has Windshield : " << (motor[i].getHasWindshield() ? "Yes" : "No") << endl;
            cout << "--------------------\n";
        }
    }

    // Cetak daftar truk
    void printTrucks() {
        cout << "Daftar Truk di Showroom " << nama << ":\n";
        if (truk.empty()) {
            cout << "Tidak ada truk di showroom.\n";
            return;
        }
        for (int i = 0; i < truk.size(); i++) {
            cout << "Truk #" << i + 1 << ":\n";
            cout << "Brand : " << truk[i].getbrand() << endl;
            cout << "Model : " << truk[i].getmodel() << endl;
            cout << "Year : " << truk[i].getyear() << endl;
            cout << "Price : " << truk[i].getprice() << endl;
            cout << "Kapasitas : " << truk[i].getkapasitas() << " ton" << endl;
            cout << "Gerbong : " << truk[i].getgerbong() << endl;
            cout << "--------------------\n";
        }
    }

    // Cetak daftar mobil
    void printCars() {
        cout << "Daftar Mobil di Showroom " << nama << ":\n";
        if (mobil.empty()) {
            cout << "Tidak ada mobil di showroom.\n";
            return;
        }
        for (int i = 0; i < mobil.size(); i++) {
            cout << "Mobil #" << i + 1 << ":\n";
            cout << "Brand : " << mobil[i].getbrand() << endl;
            cout << "Model : " << mobil[i].getmodel() << endl;
            cout << "Year : " << mobil[i].getyear() << endl;
            cout << "Price : " << mobil[i].getprice() << endl;
            cout << "Doors : " << mobil[i].getdoors() << endl;
            cout << "Fuel Type : " << mobil[i].getfuelType() << endl;
            cout << "--------------------\n";
        }
    }

    // Cetak semua kendaraan di showroom
    void printAllVehicles() {
        cout << "Showroom: " << nama << " | Lokasi: " << lokasi << "\n";
        cout << "=================================\n";
        printMotorcycles();
        cout << "\n";
        printTrucks();
        cout << "\n";
        printCars();
        cout << "=================================\n";
    }

    // Destructor
    ~Showroom() {
        cout << "Showroom " << nama << " dihapus\n";
    }
};
